jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -904,8 +905,17 @@ def _fit_model_to_spectrum(self, add_data):
             return
         models_to_fit = self._reinitialize_with_fixed()
 
-        masked_spectrum = self._apply_subset_masks(self.dataset.selected_spectrum,
+        spec = self.dataset.selected_spectrum
+        # Needs Attention:
+        # should conversion occur before or after call to _apply_subset_masks?
+        if spec.flux.unit.to_string != self.app._get_display_unit('flux'):
```

Comment:
Isn't this a method?  Is it cheaper to compare units instead of casting to string?

```suggestion
        if spec.flux.unit.to_string() != self.app._get_display_unit('flux'):
```

Reply:
Good catch! Is it worth converting the display units to astropy.Unit or keep to string comparison?

Reply:
I'm not sure, probably fine as is for now